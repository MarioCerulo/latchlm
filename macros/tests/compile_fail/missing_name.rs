#![allow(unused)]

use latchlm_core::AiModel;
use latchlm_macros::AiModel;

#[derive(AiModel)]
enum Model {
    #[model(id = "variant")]
    Variant,
}

fn main() {
    Model::Variant.as_ref();
}
