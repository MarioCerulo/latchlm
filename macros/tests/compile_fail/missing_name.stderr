error: missing #[model] attribute with name
 --> tests/compile_fail/missing_name.rs:9:5
  |
9 |     Variant,
  |     ^^^^^^^

error[E0599]: no method named `as_ref` found for enum `Model` in the current scope
  --> tests/compile_fail/missing_name.rs:13:20
   |
7  | enum Model {
   | ---------- method `as_ref` not found for this enum
...
13 |     Model::Variant.as_ref();
   |                    ^^^^^^ method not found in `Model`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `as_ref`, perhaps you need to implement it:
           candidate #1: `AsRef`
